### ISSUE TITLE
Increase the number of system calls intercepted by Hijack

### ISSUE THREAD

**Issue by Rondom** [lkl/linux#201]
From what I can see only a very limited number of system calls is actually intercepted by lkl-hijack.

I am wondering what speaks against widening the number of system calls supported. What would be the pros and cons? Would it interfere with some use cases where people still need programs to access some parts of their host system?

**Comment by Rondom** [comment]
[rump-hijack manpage](http://netbsd.gw.com/cgi-bin/man-cgi?rumphijack+3+NetBSD-7.0.1) describing some additional features useful for LKL as well

**Comment by opurdila** [comment]
Yes, interference is the main issue. Think about all the filesystem calls needed to read various files from the host system (e.g. /etc/ld.so.cache), mmap to map host libraries, etc. 

Do you have specific system calls you need supported?

**Comment by thehajime** [comment]
I am mostly thinking that it's impossible to support all system calls with a hijack library due to various limitations of hijacking.  it's a magic indeed but not panacea in my experience.   and if we want more LKL syscalls supported, we may need something different as I'm (and a rump guy is) attempting with https://github.com/libos-nuse/frankenlibc (and coupling musl-libc etc).

but you're right @Rondom, librumphijack is much flexible right now and we can *improve* as they did.  Especially specifying the scope of hijack is a good way.   I'm even thinking to share/reuse their code (i.e., librumphijack) in LKL but not tried it yet.
